### ISSUE TITLE
acceptance tests message bundler: command not found: rspec

### ISSUE THREAD

**Issue by Dan33l** [puppetlabs/pdk#535]
At root of module generated by pdk : 
```
$ pdk bundle exec rspec spec/acceptance
pdk (INFO): Using Ruby 2.4.4
pdk (INFO): Using Puppet 5.5.2
bundler: command not found: rspec
Install missing gem executables with `bundle install`
```
I can find `/opt/puppetlabs/pdk/share/cache/ruby/2.4.0/bin/rspec` but not with `2.4.4`

Even after `pdk bundle install` same error message.

pdk used is `1.6.0`.

**Comment by scotje** [comment]
@Dan33l I'm still investigating the `PATH` setup under `pdk bundle exec` to see why it's not finding the `rspec` executable, but the easiest workaround right now is probably to do:

```
$ pdk bundle binstubs rspec-core
```

and then:

```
$ pdk bundle exec bin/rspec spec/acceptance
```

Can you let me know if that works?

**Comment by Dan33l** [comment]
The workaround worked as expect on my daily workstation.

But after this, i launched a new and clean virtual machine. And on this fresh install when in ran  `pdk bundle install` the install meet an error because it was not possible to install gem `nokogiri` with the following final message : 
```
An error occurred while installing nokogiri (1.8.3), and Bundler cannot continue.
Make sure that `gem install nokogiri -v '1.8.3'` succeeds before bundling.

In Gemfile:
  beaker-module_install_helper was resolved to 0.1.7, which depends on
    beaker was resolved to 3.36.0, which depends on
      beaker-vcloud was resolved to 0.3.0, which depends on
        beaker-vmware was resolved to 0.2.0, which depends on
          rbvmomi was resolved to 1.13.0, which depends on
            nokogiri
```
And also i got this : 
```
$> pdk bundle binstubs rspec-core
pdk (INFO): Using Ruby 2.4.4
pdk (INFO): Using Puppet 5.5.2
Could not find nokogiri-1.8.3 in any of the sources
```
Even with the existing files `/opt/puppetlabs/pdk/private/ruby/2.4.4/lib/ruby/gems/2.4.0/gems/nokogiri-1.8.2/bin/nokogiri` and `/opt/puppetlabs/pdk/private/ruby/2.4.4/bin/nokogiri`

On my daily workstation, i have this gem already installed out if `pdk`. And i suppose it is used.

**Comment by scotje** [comment]
@Dan33l On your virtual machine installation had you run either `pdk validate` or `pdk test unit` on the module before running `pdk bundle install`? In order to take advantage of the pre-compiled gems that are packaged with PDK, you need to run one of those commands first before doing anything with `pdk bundle`. It's possible you might also be able to run `pdk bundle install --local` (be sure to delete any existing `Gemfile.lock` in the module first though.

Regarding the initial issue you were having, I think we have confirmed that there is a bug in the PATH setup for `pdk bundle exec` that we are now tracking in https://tickets.puppetlabs.com/browse/PDK-1073

Hopefully we can get that resolved before the next PDK release.

**Comment by Dan33l** [comment]
@scotje On the fresh installed virtual machine,  neither `pdk validate` nor `pdk test unit` works. Both fail on the step `Installing missing Gemfile dependencies.`.
I did the following commands :

1. My first command was `apt-get install pdk`.
2. My second command was `pdk validate` and got error message show in a gist : https://gist.github.com/Dan33l/c368bceacb631716ede86bac3d1cd628 .
3. My third command was `pdk test unit`and also got error message show in a gist : https://gist.github.com/Dan33l/ed36796372e2c187ec0f3612a7c6376a .

My test was done on a vagrant box running a minimal Xenial install.

**Comment by scotje** [comment]
@Dan33l Can you post a gist of your module's Gemfile and Gemfile.lock?

**Comment by Dan33l** [comment]
@scotje for sure : https://gitlab.adullact.net/adullact/puppet-wordpress/blob/master/Gemfile

**Comment by scotje** [comment]
@Dan33l I think that you might be able to pin nokogiri to the version we have packaged and get it to successfully `bundle install`. See if adding `gem "nokogiri", "= 1.8.2"` to your Gemfile (and deleting `Gemfile.lock`) allows `pdk validate` to run on your virtual machine.

**Comment by Dan33l** [comment]
Nice. `nokogiri` provided by `pdk` is now used.
Thank you.

**Comment by scotje** [comment]
Great, glad you were able to get it working!

**Comment by scotje** [comment]
The `bundler: command not found: rspec` should be resolved in the forthcoming PDK 1.6.1.

**Comment by Dan33l** [comment]
Happy news for pdk users.
